Sources/Clappr_iOS/Classes/Plugin/Playback/AVFoundationPlayback.swift

```diff
@@ -489,3 +510,45 @@ open class AVFoundationPlayback: Playback {
         Logger.logDebug("destroyed", scope: "AVFoundationPlayback")
     }
 }
+
+// MARK: - DVR
+extension AVFoundationPlayback {
+    open override var minDvrSize: Double {
+        return 60.0
+    }
+
+    open override var usingDVR: Bool {
+        guard playbackType == .live else { return false }
+        guard let currentTime = player?.currentTime().seconds else { return false }
+
+        for range in seekableTimeRanges {
+            if currentTime < range.timeRangeValue.end.seconds {
+                return true
+            }
+        }
+
+        return false
+    }
+
+    open override var seekableTimeRanges: [NSValue] {
+        guard let ranges = player?.currentItem?.seekableTimeRanges else { return [] }
+        return ranges
+    }
+
+    open override var loadedTimeRanges: [NSValue] {
+        guard let ranges = player?.currentItem?.loadedTimeRanges else { return [] }
+        return ranges
+    }
+
+    open override var supportDVR: Bool {
+        guard playbackType == .live else { return false }
+
+        for range in seekableTimeRanges {
```

Comment:
I think we can inline this _for_ using something like this:

`return seekableTimeRanges.first(where: { $0.timeRangeValue. duration.seconds > minDvrSize})  != nil`

Reply:
Done!